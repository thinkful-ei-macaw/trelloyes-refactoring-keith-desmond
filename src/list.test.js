import React from 'react';
import ReactDOM from 'react-dom';
import List from './list';
import renderer from 'react-test-renderer';

it ('renders without crashing',()=>{
    const div= document.createElement('div');
    ReactDOM.render(<List cards={[{id: 'a',title:'First card',content:'lorem ipsum'}]} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', ()=>{
    const tree=renderer.create(<List cards={[{id: 'a',title:'First card',content:'lorem ipsum'}]} />).toJSON();
    expect(tree).toMatchSnapshot();
});
