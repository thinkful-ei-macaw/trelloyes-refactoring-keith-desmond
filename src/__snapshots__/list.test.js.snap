// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  classname-list="true"
>
  <header />
  <div
    className="List-cards"
  >
    <ul>
       
      <li
        className="card"
      >
        <button
          type="button"
        >
          delete
        </button>
        <h3>
          First card
        </h3>
        <p>
          lorem ipsum
        </p>
      </li>
    </ul>
    <button
      type="button"
    >
      delete
    </button>
  </div>
</section>
`;
